src/utils.jl

```diff
@@ -1,489 +1,195 @@
-
-"""
-    max_add(inv_mode::Investment)
-
-Returns the maximum allowed added capacity of the investment mode `inv_mode` as
-`TimeProfile`.
-"""
-max_add(inv_mode::Investment) = inv_mode.max_add
-"""
-    max_add(inv_data::GeneralInvData)
-
-Returns the maximum allowed added capacity of the investment data `inv_data` as
-`TimeProfile`.
-"""
-max_add(inv_data::GeneralInvData) = max_add(investment_mode(inv_data))
-"""
-    max_add(inv_mode::Investment, t_inv)
-
-Returns the maximum allowed added capacity of the investment mode `inv_mode` investment
-period `t_inv`.
-"""
-max_add(inv_mode::Investment, t_inv) = inv_mode.max_add[t_inv]
-
-"""
-    max_add(inv_data::GeneralInvData, t_inv)
-
-Returns the maximum allowed added capacity of the investment data `inv_data` in investment
-period `t_inv`.
-"""
-max_add(inv_data::GeneralInvData, t_inv) = max_add(investment_mode(inv_data), t_inv)
-
-"""
-    min_add(inv_mode::Investment)
-
-Returns the minimum allowed added capacity of the investment mode `inv_mode` as
-`TimeProfile`.
-"""
-min_add(inv_mode::Investment) = inv_mode.min_add
-"""
-    min_add(inv_data::GeneralInvData)
-
-Returns the minimum allowed added capacity of the investment data `inv_data` as
-`TimeProfile`.
-"""
-min_add(inv_data::GeneralInvData) = min_add(investment_mode(inv_data))
-"""
-    min_add(inv_mode::Investment, t_inv)
-
-Returns the minimum allowed added capacity of the investment mode `inv_mode` in investment
-period `t_inv`.
-"""
-min_add(inv_mode::Investment, t_inv) = inv_mode.min_add[t_inv]
-
-"""
-    min_add(inv_data::GeneralInvData, t_inv)
-
-Returns the minimum allowed added capacity of the investment data `inv_data` in investment
-period `t_inv`.
-"""
-min_add(inv_data::GeneralInvData, t_inv) = min_add(investment_mode(inv_data), t_inv)
-
-"""
-    increment(inv_mode::Investment)
-
-Returns the capacity increment of the investment mode `inv_mode` as `TimeProfile`.
-"""
-increment(inv_mode::Investment) = inv_mode.increment
-"""
-    increment(inv_data::GeneralInvData)
-
-Returns the capacity increment of the investment data `inv_data` as `TimeProfile`.
-"""
-increment(inv_data::GeneralInvData) = increment(investment_mode(inv_data))
-"""
-    increment(inv_mode::Investment, t_inv)
-
-Returns the capacity increment of the investment mode `inv_mode` in investment period `t_inv`.
-"""
-increment(inv_mode::Investment, t_inv) = inv_mode.increment[t_inv]
-"""
-    increment(inv_data::GeneralInvData, t_inv)
+Calculate the discounted values used in the lifetime calculations, when the `LifetimeMode`
+is given by `PeriodLife` and `StudyLife`.
 
-Returns the capacity increment of the investment data `inv_data` in investment period `t_inv`.
+# Arguments
+- `years:`: the remaining years for calculating the discounted value. The years are
+  depending on the considered [`LifetimeMode`](@ref), using `remaining(t_inv, 𝒯)` for
+  [`StudyLife`](@ref) and `duration(t_inv)` for [`PeriodLife`](@ref).
+- `lifetime`: the lifetime of the ttype.
```

Comment:
```suggestion
- `lifetime`: the lifetime of the type.
```
Or perhaps rather `element` (see above)